sdk/proto/provider.proto

```diff
@@ -23,10 +23,16 @@ package pulumirpc;
 // ResourceProvider is a service that understands how to create, read, update, or delete resources for types defined
 // within a single package.  It is driven by the overall planning engine in response to resource diffs.
 service ResourceProvider {
+    // CheckConfig validates the configuration for this resource provider.
+    rpc CheckConfig(CheckRequest) returns (CheckResponse) {}
```

Comment:
Would it make sense to introduce new message types for the arguments and return types of these functions? It seems like it could be a future versioning hazard to tie these two methods to `Diff` and `Check` if we ever diverge. (Would we ever diverge?)

Reply:
We really ought never to diverge, as these methods underly the `Check` and `Diff` implementations of the provider registry. At the very least we would need to support all of the fields contained by `{Check,Diff}{Request,Response}`, so the new types would likely embed the others. Do you have any thoughts on additional fields we might want?

Reply:
Hard for me to speculate, but if we intend to tie the semantics of `CheckConfig` to `Check` and `DiffConfig` to `Diff` permanently, then this is fine with me. I just wasn't sure if we intended to do that, but sounds like we are.

Reply:
That was the intent of the design, yes.